sgtm/artemis/extract_value_from_stringified_json_object.tpl

```diff
@@ -0,0 +1,122 @@
+﻿___TERMS_OF_SERVICE___
+
+By creating or modifying this file you agree to Google Tag Manager's Community
+Template Gallery Developer Terms of Service available at
+https://developers.google.com/tag-manager/gallery-tos (or such other URL as
+Google may provide), as modified from time to time.
+
+
+___INFO___
+
+{
+  "type": "MACRO",
+  "id": "cvt_temp_public_id",
+  "version": 1,
+  "securityGroups": [],
+  "displayName": "Extract Value from Stringified JSON Object",
+  "description": "Takes an input of Stringified JSON Object (e.g. a document from Firestore) and returns the value based on a given key.",
+  "containerContexts": [
+    "SERVER"
+  ]
+}
+
+
+___TEMPLATE_PARAMETERS___
+
+[
+  {
+    "type": "TEXT",
+    "name": "jsonString",
+    "displayName": "Stringified JSON Object",
+    "simpleValueType": true,
+    "valueValidators": [
+      {
+        "type": "NON_EMPTY"
+      }
+    ]
+  },
+  {
+    "type": "TEXT",
+    "name": "key",
+    "displayName": "Key",
+    "simpleValueType": true,
+    "valueValidators": [
+      {
+        "type": "NON_EMPTY"
+      }
+    ]
+  },
+  {
+    "type": "TEXT",
+    "name": "default",
+    "displayName": "Default Value",
+    "simpleValueType": true
+  }
+]
+
+
+___SANDBOXED_JS_FOR_SERVER___
+
+const logToConsole = require("logToConsole");
+const JSON = require("JSON");
+
+function getValueFromJSONString(jsonString, key) {
+  
+  // Attempt to parse the JSON string
```

Comment:
I would lose these comments as they're not telling us any information that we can't see by reading the code.

Use comments to highlight things that are not obvious, but assume the reader can write JS.